Classes/Domain/Repository/StackRepository.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the package jweiland/indexnow.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace JWeiland\IndexNow\Domain\Repository;
+
+use Doctrine\DBAL\FetchMode;
+use TYPO3\CMS\Core\Database\Query\QueryBuilder;
+
+final class StackRepository
```

Comment:
I'm not a fan of "final". If you want to keep that please add an EventListener for individual modifications of the QueryBuilder or record.

Reply:
I think, if a user wants it, she has to explain, why, then we can find the solution. I can't think at this time why someone wants to adjust the stored records. The repository itself should be no official API, it is used to fulfil its job in the extension - and can be changed any time. This expresses the final. But we can tag the class as @internal - so every user who wants to extend knows the risk: it can break on updates without any further notice.